nengo/utils/compat.py

```diff
@@ -75,12 +75,24 @@ def ensure_bytes(s):
 assert TextIO
 
 
+def as_ascii(string):
+    if not PY2 and isinstance(string, bytes):  # Python 3
+        return string.decode('ascii')
+    elif not isinstance(string, str):
+        return str(string)
+    else:
+        return string
+
+
 def is_integer(obj):
     return isinstance(obj, int_types + (np.integer,))
 
 
 def is_iterable(obj):
-    return isinstance(obj, collections.Iterable)
+    if isinstance(obj, np.ndarray):
```

Comment:
No, there's a difference between Numpy scalars (e.g. `np.int32(0)`) and 0-d Numpy arrays (e.g. `np.array(0, dtype=np.int32)`).